types/index.d.ts

```diff
@@ -21,8 +21,49 @@ declare function xastscript(name: string, ...children: Children[]): Element
  */
 declare function xastscript(
   name: string,
-  attributes?: Attributes,
+  attributes?: xast.Attributes,
   ...children: Children[]
-): Element
+): xast.Element
+
+/**
+ * This namespace allows to use `xastscript` as a JSX implementation.
+ *
+ * This namespace is only used to support the use as JSX. It’s **not** intended for direct usage.
+ */
+declare namespace xastscript.JSX {
+  /**
+   * This defines the return value of JSX syntax.
+   */
+  type Element = xast.Element
+
+  /**
+   * This disallows the use of
+   */
+  type IntrinsicAttributes = never
+
+  /**
+   * This defines the prop types for known elements.
+   *
+   * For `xastscript` this defines any string may be used in combination with `xast` `Attributes`.
+   */
+  interface IntrinsicElements {
+    [key: string]: xast.Attributes & {
+      /**
+       * The prop that matches `ElementChildrenAttribute` key defines the type of JSX children, defines the children type.
+       */
+      ''?: Children
```

Comment:
Out of curiosity where did you find documentation on the key `''` part for children attribute?
I scanned https://www.typescriptlang.org/docs/handbook/jsx.html and couldn't find a reference.

Reply:
I figured this out while implementing [mini-jsx](https://gitlab.com/appsemble/mini-jsx/-/blob/master/h.ts#L474-510). Finding out about this was a mix between reading the documentation and reverse engineering `@types/react`.